text/trace/0192-messaging-semantic-conventions-spec.md

```diff
@@ -0,0 +1,628 @@
+describes. In the context of messaging systems, this means that a span should
+at the very least make clear that it refers to a messaging system, in addition
+it needs to make clear what particular [messaging operation](#operation-name)
+it refers to.
+
+Ideally, the span name also contains the destination name of the messages it
+refers to. However, a destination name should only be added to the span name
+when it is of low cardinality. This is usually the case when the destination
+name is a meaningful and manually configured name (like a manually configure
+queue or topic name), it is usually not the case if the destination name is an
+auto-generated identifier (like a conversation id or an auto-generated name for
+an anonymous destination).
+
+> The span name SHOULD consist of the name of the messaging system followed by
+> an [operation name](#operation-name). The destination name MAY be appended if
+> it is of low cardinality.
+
+##### Examples
+
+* `kafka publish shop.orders`
+* `rabbitmq receive print_jobs`
+* `AmazonSQS deliver`
+* `activemq settle`
+
+#### Operation name
+
+The following operations related to messages are covered by these semantic
+conventions:
+
+| Operation name | Description |
+|----------------|-------------|
+| `publish`      | One ore more messages are provided for publishing to an intermediary. |
+| `create`       | A message is created. |
+| `receive`      | One or more messages are requested by a consumer. |
+| `deliver`      | One or more message are passed to a consumer. |
+| `settle`       | One or more message are settled. |
+
+For further details about each of those operations refer, to the [section about trace structure](#trace-structure).
+
+#### Span kind
+
+[Span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)
+SHOULD be set according to the following table, based on the operation a span describes.
+
+| Operation name | Span kind|
+|----------------|-------------|
+| `publish`      | `PRODUCER`, if no `create` spans are present. `INTERNAL` otherwise. |
+| `create`       | `PRODUCER` |
+| `receive`      | `CONSUMER` |
+| `deliver`      | `CONSUMER` |
+| `settle`       | `INTERNAL` |
+
+Setting span kinds according to this table ensures, that span links between
+consumer and producers always go from a `PRODUCER` span on the producer side to
+a `CONSUMER` span on the consumer side. This allows analysis tools to interpret
+linked traces without the need of additional semantic hints.
+
+#### Attributes
+
+Attribute             | Type   | Required
+----------------------|--------|---------
+[`messaging.broker`](#messagingbroker) | string | Yes
+[`messaging.operation`](#messagingoperation) | string | Yes
+[`messaging.destination.name`](#messagingdestinationname) | string | For producer spans
+[`messaging.destination.template`](#messagingdestinationtemplate) | string | No
```

Comment:
having different names for destination and source could be confusing for users. E.g. I'm investigating connectivity  errors for a given broker+topic, I'd have to write something like `msg. destination.name == "foo" or msg.source.name == "foo"`.

It might be small, but I don't really understand why we need to use different terminology here. 

I suggest we try to find a name that can describe any sort of messaging destination (queues, topics, subjects). Would `entity` or  `channel` work?